File: JULY-2025/25-07-2025/p3487.go
func maxSum(nums []int) int {
	set := map[int]bool{}
	maxNum := nums[0]
	sum := 0
	for _, num := range nums {
		if num > 0 && !set[num] {
			set[num] = true
			sum += num
		}
		maxNum = max(maxNum, num)
	}
	if maxNum >= 0 {
		return sum
	} else {
		return maxNum
	}
}

File: JULY-2025/25-07-2025/p3487.erl
-spec max_sum(Nums :: [integer()]) -> integer().
max_sum(Nums) ->
    Uniq_sorted = lists:usort(Nums),    %% sort and keep uniques only
    Max = lists:max(Uniq_sorted),   
    if Max < 1 -> Max;                  %% if max is non-positive, just return it
        true -> lists:sum([Num || Num <- Uniq_sorted, Num > 0])
    end.